template_credentials_test.go

```diff
@@ -0,0 +1,71 @@
+package transloadit
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+func TestTemplateCredentials(t *testing.T) {
+	t.Parallel()
+
+	client := setup(t)
+	templateCredentialName := generateTemplateName()
+
+	templateCredentialPost := NewTemplateCredential()
+	templateCredentialPost.Name = templateCredentialName
+	templateCredentialPost.Type = "s3"
+	templateCredentialPost.Content["key"] = "xyxy"
+	templateCredentialPost.Content["secret"] = "xyxyxyxy"
+	templateCredentialPost.Content["bucket"] = "mybucket.example.com"
+	templateCredentialPost.Content["bucket_region"] = "us-east-1"
+
+	// Step 1: Create a brand new templateCredentialPost
+	id, err := client.CreateTemplateCredential(ctx, templateCredentialPost)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.DeleteTemplateCredential(ctx, id)
+	if id == "" {
+		t.Error("no templateCredentialPost id returned")
+	}
+
+	// Step 2: Retrieve new templateCredentialPost and assert it's properties
+	var templateCredential TemplateCredential
+	if templateCredential, err = client.GetTemplateCredential(ctx, id); err != nil {
+		t.Fatal(err)
+	}
+
+	if templateCredential.Name != templateCredentialName {
+		t.Error("wrong templateCredentialPost name")
+	}
+	if templateCredential.Type != "s3" {
+		t.Error("wrong templateCredentialPost type")
+	}
```

Comment:
Can we also add an assertion that the actual content of the credential is correct? For example, the bucket, key etc?